lib/types/local-time.js

```diff
@@ -27,257 +29,229 @@ const nanoSecInMillis = Long.fromNumber(1000000);
+    /**
+     * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Date} date Local date portion to extract the time passed since midnight.
+     * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the nanosecond time portion.
+     * @returns {LocalTime}
+     */
+    static fromDate(date, nanoseconds) {
+        if (typeof totalNanoseconds === "Date") {
+            throw new TypeError("Not a valid date");
+        }
+        // Use the local representation, only the milliseconds portion
+        const millis =
+            (date.getTime() + date.getTimezoneOffset() * -60000) % millisInDay;
+        return LocalTime.fromMilliseconds(millis, nanoseconds);
     }
-    return new LocalTime(new Long(value.readInt32BE(4), value.readInt32BE(0)));
-};
-
-/**
- * Compares this LocalTime with the given one.
- * @param {LocalTime} other time to compare against.
- * @return {number} 0 if they are the same, 1 if the this is greater, and -1
- * if the given one is greater.
- */
-LocalTime.prototype.compare = function (other) {
-    return this.value.compare(other.value);
-};
-
-/**
- * Returns true if the value of the LocalTime instance and other are the same
- * @param {LocalTime} other
- * @returns {Boolean}
- */
-LocalTime.prototype.equals = function (other) {
-    return other instanceof LocalTime && this.compare(other) === 0;
-};
-
-/**
- * Gets the total amount of nanoseconds since midnight for this instance.
- * @returns {Long}
- */
-LocalTime.prototype.getTotalNanoseconds = function () {
-    return this.value;
-};
 
-LocalTime.prototype.inspect = function () {
-    return this.constructor.name + ": " + this.toString();
-};
+    /**
+     * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Number} milliseconds A Number from 0 to 86,399,999.
+     * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
+     * @returns {LocalTime}
+     */
+    static fromMilliseconds(milliseconds, nanoseconds) {
+        if (typeof nanoseconds !== "number") {
+            nanoseconds = 0;
+        }
+        return new LocalTime(
+            Long.fromNumber(milliseconds)
+                .multiply(nanoSecInMillis)
+                .add(Long.fromNumber(nanoseconds)),
+        );
+    }
 
-/**
- * Returns a big-endian bytes representation of the instance
- * @returns {Buffer}
- */
-LocalTime.prototype.toBuffer = function () {
-    const buffer = utils.allocBufferUnsafe(8);
-    buffer.writeUInt32BE(this.value.getHighBitsUnsigned(), 0);
-    buffer.writeUInt32BE(this.value.getLowBitsUnsigned(), 4);
-    return buffer;
-};
+    /**
+     * Creates a new instance of LocalTime from the bytes representation.
+     * @param {Buffer} value
+     * @returns {LocalTime}
+     */
+    static fromBuffer(value) {
+        if (typeof value === "Buffer") {
```

Comment:
@ZuzaOsa so what shoul be here?

Reply:
I would say ``instanceof``, but this requires further research about the difference between ``instanceof`` and ``typeof``.